src/frontend/lib/utils.ts

```diff
@@ -282,3 +283,19 @@ export function matchPreset(
 
   return bestMatch;
 }
+
+export function isSavablePhoto(
+  photo: Photo | {draftPhotoId: string},
```

Comment:
related to other comment about the `capture` field, but ideally the `Photo` type represents the different states of a photo (draft or not), removing the need for the union here.

Reply:
Can you expand a little more what you mean/ what you envision?

Reply:
guess I'm wondering why there's a specific case of `{ draftPhotoId: string }` and why that's not a `DraftPhoto`? The `DraftPhoto` is defined as:

```ts
export interface DraftPhoto {
  // If the photo is still being captured
  capturing: boolean;
  draftPhotoId: string;
  // uri to a local full-resolution image (this is uploaded to Mapeo server)
  originalUri?: string;
  // uri to a local thumbnail image (this is uploaded to Mapeo server)
  thumbnailUri?: string;
  // uri to a local preview image
  previewUri?: string;
  // If an image is to be deleted
  deleted?: boolean;
  // If there was any kind of error on image capture
  error?: boolean;
}
```

so basically the difference is that `capturing` field not being included. my other comment is wondering about why `capturing` is being omitted. if it's an explicitly different kind of draft photo or something, i think it would make sense to update the `Photo` type so that it looks like:

```ts
// If the last variant can be a named type, that would be ideal
export type Photo = SavedPhoto | DraftPhoto | { draftPhotoId: string };
```

That way you need to update the type of the parameter for `isSavablePhoto()` here.

Reply:
AH! I see what you are saying.

This has exposed that there are actually 3 states (not 2) that a photo can be in
1. Saved photo: Is saved to the db and is attached to an observation (ideally - but could be orphaned because saving photos and observations are not atomic)
2. Draft Photo - fully processed: This is not saved to the db. And has a thumbnail, preview, and original
3. Draft Photo - in processing: This is not saved to the db. And has a promise that resolves to the original. And it in the state of finishing that promise to create the original AND is also creating the thumbnail and preview.

Before, types 2 and 3 were just considered 1 type. And I think that has caused alot of typescript ambiguity and fanagaling that hopefully exposing a third type will make a little easier

Reply:
I followed this pattern, which did simplify things.